src/fmu/ensemble/ensemble.py

```diff
@@ -199,13 +210,29 @@ def add_realizations(
             globbedpaths = glob.glob(paths)
 
         count = 0
-        for realdir in globbedpaths:
-            realization = ScratchRealization(
-                realdir,
-                realidxregexp=realidxregexp,
-                autodiscovery=autodiscovery,
-                batch=batch,
-            )
+        if use_concurrent():
+            with ProcessPoolExecutor() as executor:
+                loaded_reals = [
+                    executor.submit(
+                        ScratchRealization,
+                        realdir,
+                        realidxregexp=realidxregexp,
+                        autodiscovery=autodiscovery,
+                        batch=batch,
+                    ).result()
+                    for realdir in globbedpaths
+                ]
```

Comment:
Not really concurrent, as the `.result()` waits for the subprocess to finish for each iteration. Should rather do something like below (which is how it is done in most of the rest of this PR, so this is probably just something you forgot to fix).
```suggestion
               reals_futures = [
                    executor.submit(
                        ScratchRealization,
                        realdir,
                        realidxregexp=realidxregexp,
                        autodiscovery=autodiscovery,
                        batch=batch,
                    )
                    for realdir in globbedpaths
                ]
                loaded_reals = [x.result() for x in reals_futures]
```